### PR TITLE
Bundle A (#1677): tier-keyed model routing (#1672) + seeded-defect corpus & catch-rate benchmark (#1675)

### LINKED ISSUE #1672: Model routing ignores the risk-tier classifier — collapses to flat opus (less differentiated than native plan mode / dynamic workflows)
## Problem

Exarchos selects the dispatch **model** in the classify path (`classifyTaskCore`, `servers/exarchos-mcp/src/orchestrate/task-decomposition.ts`) from scaffolding-keyword / testLayer / deps / file-count — **decoupled from the `riskTier` the same classifier already computes**. On the real `docs/specs/` corpus the model mix collapses to `{ implementer/opus: 99, scaffolder/haiku: 1 }` — effectively everything → opus, and the one `haiku` downgrade landed on a **high-tier** task (miscalibrated).

## Why it matters

Native Claude Code **plan mode** and **dynamic workflows** already dispatch `haiku`/`sonnet`/`opus` per subtask as appropriate — they are **not** flat. So exarchos's flat-opus routing is **strictly less differentiated than the native baseline it is meant to improve on**, and it ignores the risk signal we already compute. The #1670 benchmark could not measure any "vs native" model-selection benefit because our side has no differentiation to measure — the classifier's output never reaches model selection.

## Proposed remediation

Make model selection a deliberate function of the **risk tier** (and/or task complexity), reconciled with the scaffolder/implementer split — a documented `tier → model` policy (low → cheaper, high → stronger), never routing a high-tier task to a cheap model. Exact mapping to be settled in design (`/exarchos:ideate`), validated against the invariants catalog.

## Acceptance criteria

- Model selection reads `riskTier`, not only keyword/file-count heuristics.
- The `docs/specs/` corpus no longer collapses to a single model; the mix tracks the tier distribution.
- No high-tier task is routed to a cheap model; no trivial low-tier task is forced to opus without cause.
- A test asserts the `tier → model` policy.

## Blocks / relationship to #1670

Exp 2 of #1670 (a *measured* native model-selection baseline) **cannot be concluded** until BOTH:
1. this routing defect is fixed (so exarchos actually differentiates), and
2. the Exp-2 `claude -p` harness is corrected to observe native's **real** dynamic/plan-mode routing — the current spike pins `--model`, which forces a flat result and proves nothing about native's behavior, and it drives a flat `--allowedTools Task` prompt rather than plan-mode / dynamic-workflow dispatch (the modes that actually route models).

Refs: `docs/evals/2026-07-09-1636-plan-format-corpus.md` (Dimension 1), `docs/evals/2026-07-09-1670-delegation-pipeline-empirical.md` (Exp 2).

### LINKED ISSUE #1675: Benchmark Pillar 3 — mechanical-gate catch-rate on a seeded-defect corpus (the enforcement floor)
**Pillar 3 of the verification-design validation suite** (see the epic). Benchmarks the **mechanical gates as enforcement** — `check_test_adequacy` (mutation kill-probe), `check_integration_suite`, `check_contract_drift`, `check_mock_boundary`, `check_static_analysis`. **This is the biggest gap #1670 left open.**

## Context — why this is untested

A gate's value is a **floor**, not a lift on the mean: its job is to *reject bad output* and *block an under-verified merge*. #1670's corpus produced only good output (both arms scored adequate everywhere), so every gate looked redundant — its entire value lives on the failure tail the benchmark never generated. `grade.test.ts` already shows the mutation gate discriminates a vacuous vs a genuine test on a fixture; this issue generalizes that into a real **catch-rate** benchmark across all gates.

## Experiment

- Build a **seeded-defect corpus**: for each gate, a set of inputs that *should* fail it —
  - test-adequacy / mutation → vacuous / tautological tests,
  - oracle → a dropped edge case,
  - contract-drift → a broken seam contract,
  - mock-boundary → an over-mocked / faked-what-you-don't-own boundary,
  - static-analysis → a type / lint violation,
  - integration-suite → a broad-blast regression invisible to per-task gates.
- Measure per gate: **true-positive (catch) rate** on seeded defects and **false-positive rate** on a matched set of known-good inputs.

## Acceptance criteria

- A per-gate **catch-rate + false-positive-rate** table over ≥N seeded defects and ≥N good controls each, mechanically run.
- Any gate with a **low catch rate** (misses real defects) or **high false-positive rate** (blocks good work) is flagged for redesign or removal — this is the experiment that says which gates actually earn their place in the ladder.

## Dependencies / notes

None — gates exist as `exarchos_orchestrate` actions. The seeded-defect corpus is reused by Pillar 2 (escaped-defect rate) and Pillar 4 (regression ratchet). Refs: #1670, `grade.test.ts` (existing gate-discrimination fixture).

### PR DESCRIPTION
**Pull request by rsalus**
## Summary

**Bundle A** of epic #1677 (verification-design validation benchmarks) — the fix + the enforcement floor. Two independent deliverables that together produce every artifact Bundle B consumes:

- **#1672 — Tier-keyed model routing (DR-1)**, the epic's *only* production-code change. Model selection now derives from the resolved `riskTier` instead of the agent lane alone, closing the #1670 miscalibration where ~99/100 tasks collapsed to a single model.
- **#1675 — Seeded-defect corpus + mechanical-gate catch-rate benchmark (DR-2, DR-3)**, the failure-tail substrate #1670 never produced, plus the first executed measurement of the enforcement floor.

Spec: `docs/specs/2026-07-10-1677-verification-pillar-benchmarks.md`.

### #1672 — DR-1 tier-keyed model routing (tasks 001–002)
- `config/resolve.ts` + `config/yaml-schema.ts`: new `agents.tierModels: Record<RiskTier, 'opus'|'sonnet'|'haiku'>` (in-code defaults **low→haiku, medium→sonnet, high→opus**), overridable via a new `.exarchos.yml agents.tier-models` key. Config-resolution-time `validateTierModels` enforces the settled guard shape: **rejects** non-monotone tables and `high→haiku`; **allows** `high→sonnet` (Open Question 2, settled at plan-review). Structured, cell-naming config errors matching the existing envelope style.
- `orchestrate/prepare-delegation.ts`: `resolveModelForTask(agent, riskTier, config)` applied in `classifyTask` on top of `classifyTaskCore` (which still owns agent/complexity/effort). The planner-stamped tier (#1669) now drives the model. Per-agent `agents.models` retained for non-dispatch surfaces (reviewer/fixer dispatch, agent generation). `TaskClassification.recommendedModel` type unchanged → registered outputSchema untouched, CLI/MCP parity holds (INV-2).
- `evals/benchmarks/plan-format-corpus.ts`: refreshed a now-stale Dimension-1 comment that claimed model routing is "independent of `riskTier`".

### #1675 — Seeded-defect corpus + catch-rate benchmark (tasks 003–005)
- `evals/benchmarks/seeded-defects/`: 6 defect classes × (5 seeded + 5 controls) = **60 inert JSON fixtures** (never compiled TS); tiers **derived by the production `deriveRiskTier`/`deriveBoundaryTouching`** from real file paths (anti-pinning contract). Single `loadSeededCorpus(gateClass?)` loader. Dropped-edge-case class ships its hidden-oracle detector (no production gate catches it — the escaped-defect substrate for Bundle B's DR-5 replay, deliberately excluded from the gate catch-rate table; declared deviation from #1675's six-row table).
- Fixtures directory excluded from the MCP server `tsconfig` + eslint config so intentionally type/lint-broken fixture content cannot fail repo CI.
- `catch-rate-driver.ts`: drives the five real mechanical gate handlers over every fixture in disposable worktrees with an **ephemeral event store (never the project store)**; records per-cell verdict, wall-clock ms, and payload tokens (the DR-5 cost columns); crash/timeout → explicit `invalid` cell.
- `docs/evals/data/2026-07-10/gate-catch-rate.csv`: executed output, 60 rows, `stampProvenance`-stamped `source=measured`.
- `docs/evals/2026-07-10-1675-gate-catch-rate.md` + `generate_charts.py`: #1670-standard findings doc with per-gate catch-rate/FPR + cost table, verdict flags, Reproduce block, pinned provenance, and honest-scope section.

## Test Plan

- **Chain A (routing):** scoped suites green independently (`resolve.test.ts` + `prepare-delegation.test.ts` = 160 passed), typecheck clean. The executed corpus assertion `PrepareDelegation_StampedCorpus_ModelMixTracksTierDistribution` runs the real `classifyTask` over the real 140-task stamped `docs/specs/` corpus: pre-change opus:139/haiku:1 (collapsed) → post-change **haiku:20 / sonnet:70 / opus:50** (tracks the tier distribution, no single-model collapse). Includes `ResolveConfig_HighTierSonnet_Accepted` (OQ2) and `ResolveConfig_HighTierHaiku_Rejected`.
- **Chain B (corpus):** seeded-defects suite green (14 passed), typecheck clean, fixtures confirmed excluded from tsc + eslint. Executed catch-rate run: all five gates **100% TPR / 0% FPR / 0 invalid**; dropped-edge oracle 5/5 defects, 5/5 controls cleared. Chart regenerates byte-identically from the committed CSV.
- **Integrated result:** merged feature branch typechecks clean; full MCP-server suite passes with only the pre-existing `merge-orchestrate.*` + `store.race` baseline failures (documented red on `main`), zero new failures in changed code.

> **Reviewer note:** the 100%/0% catch-rate across every gate warrants adversarial scrutiny per DR-8 — is the corpus genuinely stressing the failure tail, or tuned to what the gates catch? Bounded with N=5 Wilson CIs + a mock-boundary precision caveat in the findings doc; gates were driven directly (raw verdicts) to avoid the workflow-severity wrapper confounding the measurement (documented in the driver header).

Closes #1672
Closes #1675
Part of #1677 (Bundle B — pillars 1/2/4 + verdict synthesis — follows on a stacked PR)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by rsalus** [comment]
## Adversarial review — verdict: PASS (no merge blockers)

A fresh-context adversarial reviewer examined the full Bundle A diff, with the central mandate to determine whether the **100% TPR / 0% FPR / 0-invalid** catch-rate is an honest measurement or an artifact.

**Central finding — honest, but a floor on the easy center (not the tail).** The number is a real measurement: gates driven through their real handlers, an ephemeral event store (never the project store), a provenance-stamped `source: measured` CSV, and tiers **genuinely derived** by the production classifier (the raw fixture JSON carries no tier fields at all — anti-pinning is *structural*). But it measures the deliberately-clean center of each defect class — canonical clean-hit defects + controls authored to dodge each gate's known false-positive trigger — so 100%/0% is close to *guaranteed-by-construction*, not a probe of the precision/recall failure tail. Every one of those artifacts was already disclosed in the findings doc (Wilson-CI floor verdict, mock-boundary precision caveat, the control-tuning admission), which is why it passes.

**DR-1** (guard + tier-keyed wiring) verified correct, exhaustively tested, and INV-2-safe; the corpus assertion is a genuine kill-probe (independently confirmed — reverting the fix turns it red).

### Findings applied in `261cfcde` (all non-blocking)
- **MEDIUM** — findings-doc headline reframed as *floor on the easy center, not the failure tail*, and now names the obligation that the **DR-9 "gates earn their keep" verdict must rest on Bundle B's DR-4 harder corpus**, not this floor.
- **MEDIUM** — `check_mock_boundary` row marked **detection-only / advisory** (it forces `passed=true` at the default config, so its "catch" enforces nothing), distinct from the four enforce-capable gates.
- **LOW** — added a **structural** anti-pinning test assertion (raw fixture JSON contains no `riskTier`/`boundaryTouching` keys) — real kill-power vs. the circular re-derivation check.
- **LOW** — commented `classifyTaskCore.recommendedModel` as a placeholder always overridden by the tier policy (guards against a future consumer reintroducing the #1670 collapse).

Supporting gates: static-analysis PASS; MCP typecheck clean; full suite 26 baseline-only failures (zero new); DR-1 kill-probe confirmed red-on-revert.

File: servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/catch-rate-driver.test.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    expect(report.eventStoreDir.startsWith(tmpRoot)).toBe(true);
+    expect(report.eventStoreDir.startsWith(os.tmpdir())).toBe(true);
+    expect(report.eventStoreDir.startsWith(REPO_ROOT)).toBe(false);
+    // The store dir was actually created (events had somewhere ephemeral to go).
+    expect(fs.existsSync(report.eventStoreDir)).toBe(true);
+    // No project-store artifact (`.exarchos`) was created inside the repo by the run.
+    // (The driver only ever writes under tmpRoot.)
+    expect(report.eventStoreDir.includes(path.join('.claude', 'worktrees')) &&
+      report.eventStoreDir.startsWith(REPO_ROOT)).toBe(false);
+  });
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/catch-rate-driver.test.ts:146]: Fixed in `80a89205`. Dropped the always-false `&& startsWith(REPO_ROOT)` (that operand was already asserted `false` above, making the whole expression vacuously false), so the `.claude/worktrees` check now stands alone as a real assertion, and corrected the stale `.exarchos` comment. Good catch.

File: servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/catch-rate-driver.ts [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+      correct,
+      wallClockMs,
+      payloadChars,
+      payloadTokens: estimateTokens(payloadChars),
+      oracleDetected: '',
+      note,
+    };
+  } finally {
+    fs.rmSync(worktree, { recursive: true, force: true });
+  }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/catch-rate-driver.ts:394]: Fixed in `80a89205`. `materializeFixture` now fails fast on any non-zero git exit (new `FixtureMaterializationError`), and the driver catches it to record an explicit `invalid` cell (`materialize-failed:…`) — kept distinct from the `handler-threw` path, so a partial worktree can never produce a trustworthy-looking verdict (DR-8). Added a regression test: `CatchRateDriver_GitMaterializeFailure_EmitsInvalidRecordNotVerdict`.

File: servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/fixtures/dropped-edge-case.json [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            "expected": "hello-world"
+          },
+          {
+            "args": [
+              "Hi"
+            ],
+            "expected": "hi"
+          }
+        ]
+      }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/fixtures/dropped-edge-case.json:140]: Fixed in `80a89205`. Added a leading-separator case `"!Hello" → "hello"` to the slugify oracle for both defect-04 and its matching control-04, so the oracle now exercises the full leading/trailing-trim mechanism. Verified: the defect impl yields `-hello` (correctly detected), while the control's `replace(/^-+|-+$/g,'')` yields `hello` (stays clean), so the 0% FPR is preserved.

File: servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/fixtures/integration-suite.json [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+      },
+      "head": {
+        "package.json": "{\n  \"name\": \"seeded-fixture\",\n  \"version\": \"1.0.0\",\n  \"private\": true,\n  \"type\": \"module\",\n  \"scripts\": {\n    \"test:run\": \"node print-report.mjs\"\n  }\n}\n",
+        ".npmrc": "loglevel=silent\n",
+        "print-report.mjs": "import { readFileSync } from 'node:fs';\nprocess.stdout.write(readFileSync(new URL('./report.json', import.meta.url), 'utf-8'));\n",
+        "report.json": "{\n  \"numFailedTestSuites\": 3,\n  \"numFailedTests\": 2,\n  \"numTotalTests\": 30,\n  \"testResults\": [\n    {\n      \"name\": \"src/x.test.js\",\n      \"status\": \"failed\",\n      \"assertionResults\": [\n        {},\n        {}\n      ]\n    },\n    {\n      \"name\": \"src/y.test.js\",\n      \"status\": \"failed\",\n      \"assertionResults\": []\n    }\n  ]\n}\n",
+        "src/a.js": "export const feature11 = () => 11;\n",
+        "src/b.js": "export const feature12 = () => 12;\n",
+        "src/c.js": "export const feature13 = () => 13;\n"
+      }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply servers/exarchos-mcp/src/evals/benchmarks/seeded-defects/fixtures/integration-suite.json:92]: Fixed in `80a89205`. Set `numFailedTestSuites: 2` to match the stated mechanism (2 failed tests in one suite + 1 load failure). The gate's parser computes `loadFailures = failedSuites − suitesWithFailedTests = 2 − 1 = 1`, which is now internally consistent, and it still FAILs the fixture (catch-rate preserved). Regenerated the committed CSV + chart (shape unchanged: 100% TPR / 0% FPR / 0 invalid).
